test/imath/bind/imath_vec.cpp

```diff
@@ -50,13 +57,29 @@ template <class T> class Vec3 {
     constexpr static T baseTypeSmallest();
     constexpr static T baseTypeEpsilon();
 
+    template <class S> IMATH_HOSTDEVICE void setValue (S a, S b, S c);
+    */
+
 } CPPMM_VALUETYPE;
 
 // These type aliases will cause the class template to be monomorphized with
 // the given template parameter as if it had been explicitly declared
 using V3f = Vec3<float>;
+V3f v3f;
 using V3i = Vec3<int>;
-
-} // namespace IMATH_INTERNAL_NAMESPACE
+V3i v3i;
 
 } // namespace cppmm_bind
+
+namespace IMATH_INTERNAL_NAMESPACE {
+
+// in order for clang to see the code for template functions that depend on
+// another template parameter list, we have to explicitly specialize them for
+// all the types we're interested in like this:
+// yuck - just realised that for matching we're going to need to do this for
+// the binding version as well.
+template <>
+template <>
+IMATH_HOSTDEVICE void Vec3<float>::setValue(float a, float b, float c);
```

Comment:
Maybe a macro will make it easier.

Reply:
Yeah I’m just going to punt this one down the road. Thinking about it, using the macro would be almost as verbose and confusing as the raw declarations. 

What /would/ be much simpler is a custom DSL we transform to C++ at binding time. But let’s save that for version 2 maybe...